src/App.tsx

```diff
@@ -1,14 +1,53 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Select } from './utils/Select';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterOption, setFilterOption] = useState(Select.All);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        // eslint-disable-next-line no-console
+        console.log('setTodos error');
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  const filterByFilterOption = useCallback((todosToFilter: Todo[]): Todo[] => {
+    return todosToFilter.filter(todo => {
+      switch (filterOption) {
+        case Select.Active: return !todo.completed;
+        case Select.Completed: return todo.completed;
+        case Select.All:
+        default: return true;
+      }
+    });
+  }, [todos, filterOption]);
+
+  const filterByQuery = useCallback((todosToFilter: Todo[]) => {
+    return todosToFilter.filter(todo => todo.title.toLowerCase().includes(query.toLowerCase().trim()));
+  }, [todos, query]);
```

Comment:
Better to move these functions outside the component and use them as helpers, `useCallback` is redundant in this case too